lib/ignored-paths.js

```diff
@@ -92,21 +69,81 @@ function IgnoredPaths(patterns) {
  * @returns {IgnoredPaths} object, with patterns loaded from the ignore file
  */
 IgnoredPaths.load = function(options) {
-    var patterns;
+    var ignoredPaths = new IgnoredPaths();
+    ignoredPaths.loadFile(options);
+    return ignoredPaths;
+};
 
-    options = options || {};
+/**
+ * Reset IgnoredPaths
+ * @returns {void}
+ */
+IgnoredPaths.prototype.reset = function() {
+    this.baseDir = null;
+    this.ig = new ignore.Ignore({
+        ignore: this.defaultPatterns
+    });
+
+    this.ig.on("warn", function(e) {
+        var originInfo = (e.data && e.data.origin) ? " (" + e.data.origin + ")" : "";
+        console.info("IgnoredPaths: " + e.message + originInfo);
+    });
+};
 
-    if (options.ignore) {
-        patterns = loadIgnoreFile(options.ignorePath || findIgnoreFile());
-    } else {
-        patterns = [];
-    }
+/**
+ * Load wrapper for IgnorePaths instances
+ * @param {Object} options object containing 'ignore' and 'ignorePath' properties
+ * @returns {void}
+ */
+IgnoredPaths.prototype.loadFile = function(options) {
+
+    var ignorePath;
+
+    if (options) {
+
+        if (options.ignore) {
+
+            if (options.ignorePattern) {
+                this.ig.addPattern(options.ignorePattern);
+            }
+
+            if (options.ignorePath) {
+                try {
+                    fs.statSync(options.ignorePath);
+                    ignorePath = options.ignorePath;
+                } catch (e) {
+                    e.message = "Cannot read ignore file: " + options.ignorePath + "\nError: " + e.message;
+                    throw e;
+                }
+            } else {
+                ignorePath = findIgnoreFile();
+                try {
+                    fs.statSync(options.ignorePath);
+                } catch (e) {
+                    return this;
+                }
+            }
+
+            this.baseDir = path.dirname(ignorePath);
+            this.ig.addIgnoreFile(ignorePath);
+
+        }
+
+        this.options = options;
 
-    if (options.ignorePattern) {
-        patterns.push(options.ignorePattern);
     }
 
-    return new IgnoredPaths(patterns);
+    return this;
+
+};
+
+/**
+ * Get list ignore rules
+ * @returns {IgnoreRule} array, list of ignore patterns
+ */
+IgnoredPaths.prototype.getRules = function() {
```

Comment:
The list of rules was [used by cli-engine as ignore pattern](https://github.com/eslint/eslint/blob/master/lib/cli-engine.js#L510) to reduce the number of `executeOnFile` calls. My concern was to expose an interface [similar to direct access to the pattern list](https://github.com/eslint/eslint/blob/master/lib/cli-engine.js#L499). Using an object with `name` property looked reasonable, so only one line had to be changed in external files.

The better approach in my opinion would be to use ignore's `ig.filter(paths);` method on all glob matches at once (without passing in an ignore list) and then [run executeOnFile without using IgnoredPaths](https://github.com/eslint/eslint/blob/master/lib/cli-engine.js#L558).

The second reason this method is exposed are unit tests. We need access to the raw rules for testing, but the linter complains about the _ in `this.ig._rules` :smirk_cat: That's why the next two lines exist.

If we go with the changes to `cli-engine.js`, this method can be moved to a function in the tests file.